lib/spack/spack/filesystem_view.py

```diff
@@ -67,44 +67,74 @@ def view_copy(src, dst, view, spec=None):
-
-        prefix_to_projection = collections.OrderedDict(
-            {spec.prefix: view.get_projection_for_spec(spec)}
-        )
-
-        for dep in spec.traverse():
-            if not dep.external:
-                prefix_to_projection[dep.prefix] = view.get_projection_for_spec(dep)
-
-        if spack.relocate.is_binary(dst):
-            spack.relocate.relocate_text_bin(binaries=[dst], prefixes=prefix_to_projection)
-        else:
-            prefix_to_projection[spack.store.layout.root] = view._root
-            prefix_to_projection[orig_sbang] = new_sbang
-            spack.relocate.relocate_text(files=[dst], prefixes=prefix_to_projection)
-        try:
-            stat = os.stat(src)
-            os.chown(dst, stat.st_uid, stat.st_gid)
-        except OSError:
-            tty.debug("Can't change the permissions for %s" % dst)
+    is_link = os.path.islink(src)
+
+    # Just copy the file when not a link, whether relocating it or not.
+    if not is_link:
+        shutil.copy2(src, dst)
+
+    # If we're not relocating (no spec), we can
+    # early exit; don't forget to create a symlink if necessary.
+    # Note: replace with copy2(src, dst, follow_symlinks=False) when
+    # dropping Python 2.7.
+    if not spec or spec.external:
+        if is_link:
+            os.symlink(os.readlink(src), dst)
+        return
+
+    # Now collect some paths for relocation purposes.
+    prefix_to_projection = collections.OrderedDict(
+        {spec.prefix: view.get_projection_for_spec(spec)}
+    )
+
+    for dep in spec.traverse():
+        if not dep.external:
+            prefix_to_projection[dep.prefix] = view.get_projection_for_spec(dep)
+
+    # Do relocation on the link
+    if is_link:
+        # Note: we're doing literal text replacement, so using normpath is not
+        # an option (and also, normpath should be avoided since it's incorrect
+        # for paths that contain a symlinked dir followed by ..)
+        link_dst = os.readlink(src)
+        # The majority of the cases is relative links, e.g. libz.so -> libz.so.1.2.3
+        # and we copy these verbatim, without checking if they leave the prefix
+        # through multiple `..`'s -- this should be improved.
+        # Absolute links are relocated by literal string replacement -- this
+        # isn't perfect either...
+        if os.path.isabs(link_dst):
+            for (prefix, projection) in prefix_to_projection.items():
+                if link_dst.startswith(prefix):
+                    link_dst = link_dst.replace(prefix, projection, 1)
+                    break
+        os.symlink(link_dst, dst)
+        return
+
+    # This is vestigial code for the *old* location of sbang. Previously,
+    # sbang was a bash script, and it lived in the spack prefix. It is
+    # now a POSIX script that lives in the install prefix. Old packages
+    # will have the old sbang location in their shebangs.
+    # TODO: Not sure which one to use...
```

Comment:
I don't think this is vestigial -- it's necessary for handling packages in the (very) old format).

Reply:
Ok ignore this, I just saw that the comment predates this PR.